template.yaml

```diff
@@ -11,17 +11,50 @@ Globals:
   Function:
     Timeout: 60
 
+
+Parameters:
+  Environment:
+    Description: environment name
+    Default: dev
+    Type: String
+    AllowedValues:
+      - dev
+      - staging
+      - integration
+      - prod
+    ConstraintDescription: specify dev, staging, integration or prod for environment
+
+
 Resources:
 
   KBVSessionApi:
     Type: AWS::Serverless::Api
     Properties:
-      StageName: Dev
-      Auth:
-        InvokeRole: CALLER_CREDENTIALS
+      StageName: !Ref Environment
+      DefinitionBody:
+        openapi: "3.0.1" # workaround to get `sam validate` to work
+        paths: # workaround to get `sam validate` to work
+          /foo:
```

Comment:
I can see that this resource doesn't get created, which is great. Could we rename it to `never-created` to provide that reassurance?